### ISSUE TITLE
Generted client proxy is not working with RESTEasy 

### ISSUE THREAD

**Issue by amreladawy** [mulesoft-labs/raml-for-jax-rs#438]
Using RestEasy with the generated Rest client is failing with the following error.
```
ClassCastException : class org.jboss.resteasy.client.jaxrs.engines.ManualClosingApacheHttpClient43Engine$2 cannot be cast to class MyResponse
```
The reason is that the generated ResponseDelegate inherits `javax.ws.rs.core.Response`. RESTEasy will return `ManualClosingApacheHttpClient43Engine` and it fails casting to MyReponse class.

How to get the generated code to work using RESTEasy?

**Comment by jpbelang** [comment]
Looking into it.

But, from the top of my head, does RestEasy implement jaxrs ?

**Comment by amreladawy** [comment]
Yes, it does. It is like Jersey. Actually when I switched to Jersey, the deserialisation is failing also, but I am going to list this in another issue. 

Here is the StackOverflow question. https://stackoverflow.com/questions/61431765/classcastexception-class-org-jboss-resteasy-client-jaxrs-engines-manualclosing
